### PR TITLE
Trello-684 Allow using both json-file and journald

### PR DESCRIPTION
**Pull request by bmcelvee**
https://trello.com/c/O2NQB9lR/684-document-allow-using-both-json-file-and-journald-loggingepic-ois-agl-perf

https://trello.com/c/jmVrTI3q/522-5-allow-using-both-json-file-and-journald-loggingepic-ois-agl-perf?menu=filter&filter=label:documentation

### CODE REVIEW

**Comment by bmcelvee** [comment]
@richm PTAL. I might have removed more than I needed to from what was listed in the Trello card, but I can easily replace that content. 

https://trello.com/c/jmVrTI3q/522-5-allow-using-both-json-file-and-journald-loggingepic-ois-agl-perf?menu=filter&filter=label:documentation

Thanks!

File: install_config/aggregate_logging.adoc [hunk 1]
```diff
@@ -134,7 +134,7 @@ endif::openshift-enterprise[]
 |`openshift_logging_use_ops`
 |If set to `true`, configures a second Elasticsearch cluster and Kibana for
 operations logs. Fluentd splits logs between the main cluster and a cluster reserved
-for operations logs (which consists of *_/var/log/messages_* on nodes and the logs from the
+for operations logs (which consists of the logs from the
 projects *default*, *openshift*, and *openshift-infra*). This means a second Elasticsearch
```
Comment (richm) [inline install_config/aggregate_logging.adoc]: `projects *default*, *openshift*, and *openshift-infra*`, as well as docker, openshift, and system logs from the journal).

File: install_config/aggregate_logging.adoc [hunk 2]
```diff
@@ -442,6 +438,17 @@ remove all logging persistent data including PVC, set
...
+|
+
+Permitted values are `file` and `journald`. If `file`, use `/var/log/messages` as the
+operations/system log source. If `journald`, use `journald`. The default should be
+`journald`. This will override settings applied to `USE_JOURNAL/use_journal`.
+
+If none of the previously listed parameters are set, expect the following behavior.
+System/operations logs are read from `journald`, and container/application logs
+should be read from the settings applied to Docker `--log-driver`.
+
```
Comment (richm) [inline install_config/aggregate_logging.adoc]: There is no `openshift_logging_ops_log_source`.  System logs are always read from the journal now.  For container logs, fluentd will figure out which log driver docker is using - json-file or journald - and will read the logs from that source, automatically.

File: install_config/aggregate_logging.adoc [hunk 3]
```diff
@@ -753,7 +760,7 @@ xref:../install_config/install/advanced_install.adoc#configuring-node-host-label
 [[fluentd-use-journald]]
 *Having Fluentd Use the Systemd Journal as the Log Source*
 
-By default, Fluentd reads from *_/var/log/messages_* and
+By default, Fluentd reads from `journald` and
```
Comment (richm) [inline install_config/aggregate_logging.adoc]: Fluentd will always use the `journald` as the system log source.  These are log messages from the operating system, from docker, and from openshift.  For container logs, Fluentd will determine which log driver docker is using - json-file or journald - and will automatically read the logs from that source.

**Comment by bmcelvee** [comment]
Thanks, @richm! Does this make a bit more sense?

File: install_config/aggregate_logging.adoc [hunk 4]
```diff
@@ -750,32 +747,23 @@ As part of the OpenShift cluster installation, it is recommended that you add th
...
+messages from the operating system, from Docker, and from OpenShift. For
+container logs, Fluentd will determine which log driver Docker is using,
+`json-file` or `journald`, and will automatically read the logs from that source.
 
-By default, Fluentd reads from *_/var/log/messages_* and
-*_/var/log/containers/<container>.log_* for system logs and container logs,
-respectively. You can instead use the systemd journal as the log source. There
-are three inventory parameters available:
+// [[fluentd-use-journald]]
+// *Having Fluentd Use the Systemd Journal as the Log Source*
```
Comment (richm) [inline install_config/aggregate_logging.adoc]: Does the leading `//` mean this section is commented out?

Comment (bmcelvee) [reply install_config/aggregate_logging.adoc]: It does. I wasn't certain that content was still applicable, but didn't want to delete it just in case.

**Review by richm — APPROVED** [review]
lgtm

**Comment by bmcelvee** [comment]
@openshift/team-documentation please peer review 🍁

File: install_config/aggregate_logging.adoc [hunk 5]
```diff
@@ -133,11 +133,12 @@ endif::openshift-enterprise[]
...
-for operations logs (which consists of *_/var/log/messages_* on nodes and the logs from the
-projects *default*, *openshift*, and *openshift-infra*). This means a second Elasticsearch
-and Kibana are deployed. The deployments are distinguishable by the *-ops* suffix included
-in their names and have parallel deployment options listed below.
+operations logs. Fluentd splits logs between the main cluster and a cluster
+reserved for operations logs, which consists of the logs from the projects
+*default*, *openshift*, and *openshift-infra*, as well as Docker, OpenShift, and
+system logs from the journal. This means a second Elasticsearch and Kibana are
+deployed. The deployments are distinguishable by the *-ops* suffix included in
+their names and have parallel deployment options listed below.
```
Comment (mburke5678) [inline install_config/aggregate_logging.adoc]: Link to the deployment options "below"?

File: install_config/aggregate_logging.adoc [hunk 6]
```diff
@@ -133,11 +133,12 @@ endif::openshift-enterprise[]
...
 |If set to `true`, configures a second Elasticsearch cluster and Kibana for
-operations logs. Fluentd splits logs between the main cluster and a cluster reserved
-for operations logs (which consists of *_/var/log/messages_* on nodes and the logs from the
-projects *default*, *openshift*, and *openshift-infra*). This means a second Elasticsearch
-and Kibana are deployed. The deployments are distinguishable by the *-ops* suffix included
-in their names and have parallel deployment options listed below.
+operations logs. Fluentd splits logs between the main cluster and a cluster
+reserved for operations logs, which consists of the logs from the projects
+*default*, *openshift*, and *openshift-infra*, as well as Docker, OpenShift, and
+system logs from the journal. This means a second Elasticsearch and Kibana are
```
Comment (mburke5678) [inline install_config/aggregate_logging.adoc]: Do we want to say "a second Elasticsearch and Kibana instance/pod/service/etc" ? Is it one instance for both E and K or one each? Or, will users understand?

File: install_config/aggregate_logging.adoc [hunk 7]
```diff
@@ -750,32 +747,23 @@ As part of the OpenShift cluster installation, it is recommended that you add th
 Fluentd node selector to the list of persisted
 xref:../install_config/install/advanced_install.adoc#configuring-node-host-labels[node labels].
 
-[[fluentd-use-journald]]
-*Having Fluentd Use the Systemd Journal as the Log Source*
+Fluentd uses `journald` as the system log source. These are log
+messages from the operating system, from Docker, and from OpenShift. For
```
Comment (mburke5678) [inline install_config/aggregate_logging.adoc]: s/from the operating system, from Docker, and from OpenShift/from the operating system, Docker, and OpenShift ??

File: install_config/aggregate_logging.adoc [hunk 8]
```diff
@@ -750,32 +747,23 @@ As part of the OpenShift cluster installation, it is recommended that you add th
...
+container logs, Fluentd will determine which log driver Docker is using,
+`json-file` or `journald`, and will automatically read the logs from that source.
```
Comment (mburke5678) [inline install_config/aggregate_logging.adoc]: s/will determine/determines
s/ will automatically/automatically

Comment (mburke5678) [inline install_config/aggregate_logging.adoc]: s/will determine/determines
s/will automatically read/automatically reads

**Comment by mburke5678** [comment]
@richm A few nits. LGTM otherwise

**Comment by bmcelvee** [comment]
@mburke5678 Thanks!

**Comment by bmcelvee** [comment]
[rev_history]
|xref:../install_config/aggregate_logging.adoc#install-config-aggregate-logging[Aggregating Container Logs]
|Upfated parameters in the
xref:../install_config/aggregate_logging.adoc#aggregate-logging-ansible-variables[Specifying Logging Ansible Variables] section and updated log system log source content in the xref:../install_config/aggregate_logging.adoc#aggregated-fluentd[Fluentd] section.
%

**Pull request merged**